analyses/methylation-summary/05-create-methylation-summary-table.R

```diff
@@ -0,0 +1,99 @@
+# Create Pediatric OpenTargets methylation summary table that will be utilized
+# with OPenPedCan plotting API and displayed on the NCI MTP portal
+
+# Eric Wafula for Pediatric OpenTargets
+# 03/18/2022
+
+# Load libraries
+suppressPackageStartupMessages(library(data.table))
+suppressPackageStartupMessages(library(tidyverse))
+suppressPackageStartupMessages(library(ids))
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# establish base dir
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Set path to module and results directories
+data_dir <- file.path(root_dir, "data")
+analyses_dir <- file.path(root_dir, "analyses")
+module_dir <- file.path(root_dir, "analyses", "methylation-summary")
+gene_match_input_dir <- file.path(analyses_dir,  "gene_match/input")
+results_dir <- file.path(module_dir, "results")
+
+# Creating Pediatric OpenTargets methylation summary table
+message("==========================================================")
+message("Creating Pediatric OpenTargets methylation summary table")
+message("==========================================================\n")
+
+# Create methylation summary table by merging methylation array probe quantiles 
+# and beta-tpm correlations tables
+message("Merging probe-level quantiles and beta-tpm correlations...\n")
+beta_tpm_correlations <- data.table::fread(
+  file.path(results_dir, "methyl-probe-beta-tpm-correlations.tsv.gz"),
+  showProgress = FALSE) %>% 
+  tibble::as_tibble()
+summary_table <- data.table::fread(
+  file.path(results_dir, "methyl-probe-beta-quantiles.tsv.gz"), 
+  showProgress = FALSE)  %>% 
+  tibble::as_tibble() %>% 
+  dplyr::left_join(beta_tpm_correlations, by = c("Probe_ID", "Dataset", "Disease"))
+rm(beta_tpm_correlations)
+
+# Add GENCODE version 38 (Ensembl 104) gene symbols and Ensembl IDs probe 
+# annotation to the the methylation summary table
+message("Including gene symbols and Ensembl probe annotations ...\n")
+summary_table <- data.table::fread(
+  file.path(results_dir, "methyl-probe-annotations.tsv.gz"),
+  showProgress = FALSE) %>% 
+  tibble::as_tibble() %>% 
+  dplyr::right_join(summary_table, by = "Probe_ID")
+
+# Add PMTL disgnation corresponding to GENCODE version 38 (Ensembl 104) gene
+# symbols and Ensembl IDs
+message("Including PMTL disgnation for corresponding Ensembl IDs ...\n")
+pmtl_ensembl_ids <- readr::read_tsv(file.path(gene_match_input_dir, 
+                                              "PMTL_v1.1.tsv"),
+                           show_col_types = FALSE) %>% 
```

Comment:
```suggestion
pmtl_ensembl_ids <- readr::read_tsv(file.path(data_dir, 
                                              "ensg-hugo-pmtl-mapping.tsv"),
                           show_col_types = FALSE) %>% 
```
should change this to use the file in `data/`

Reply:
fixed